tests/test_env.py

```diff
@@ -1,9 +1,32 @@
 import pytest
 
-from game_mechanics import Env
+from poker.texas_holdem import DeltaEnv
+from pettingzoo.classic import texas_holdem_v4
 
 
-def test_env():
-    """Take me out in real games."""
-    with pytest.raises(NotImplementedError):
-        env = Env()
+def raiser(state, legal_moves):
+    """Always raises"""
+    return 1
+
+
+def test_fold_looses():
+    env = DeltaEnv(texas_holdem_v4.env(), raiser, verbose=False, render=False)
+    env.reset()
+    # Raise
+    _, reward, done, _ = env.step(1)
+    assert reward == 0
+    # Fold
+    _, reward, done, _ = env.step(2)
+    assert reward < 0
+
+
+def test_illegal_move():
+    env = DeltaEnv(texas_holdem_v4.env(), raiser, verbose=False, render=False)
+    env.reset()
+    # Raise
+    _, reward, done, _ = env.step(1)
+    assert reward == 0
+    # Cant call after an opponent raise
+    _, reward, done, _ = env.step(3)
+    assert done
+    assert reward == -10
```

Comment:
womp. This throws an error:

```============================================================================== FAILURES ==============================================================================
_________________________________________________________________________ test_illegal_move __________________________________________________________________________

    def test_illegal_move():
        env = DeltaEnv(texas_holdem_v4.env(), raiser, verbose=False, render=False)
        env.reset()
        # Raise
        _, reward, done, _ = env.step(1)
        assert reward == 0
        # Cant call after an opponent raise
>       _, reward, done, _ = env.step(3)

tests/test_env.py:30: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
delta_poker/env_wrapper.py:161: in step
    reward = self._step(move)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <delta_poker.env_wrapper.DeltaEnv object at 0x7ffbd57bf040>, move = 3

    def _step(self, move: int) -> float:
    
        assert not self.done, "Game is done! Please reset() the env before calling step() again"
>       assert move in self.legal_moves, f"{move} is an illegal move"
E       AssertionError: 3 is an illegal move

delta_poker/env_wrapper.py:142: AssertionError
====================================================================== short test summary info =======================================================================
FAILED tests/test_env.py::test_illegal_move - AssertionError: 3 is an illegal move
```

Reply:
ah it hasn't been updated post-masking strategy